### PR TITLE
Adds checking of reserved keywords against team names.

### PR DESCRIPTION
**Pull request by LefsFlarey**
This fixes and solves #3789.

### CODE REVIEW

File: models/org_team.go [hunk 1]
```diff
@@ -196,7 +196,11 @@ func (t *Team) RemoveRepository(repoID int64) error {
 
 // NewTeam creates a record of new team.
 // It's caller's responsibility to assign organization ID.
-func NewTeam(t *Team) error {
+func NewTeam(t *Team) (err error) {
+	if err = IsUsableUsername(t.Name); err != nil {
```
Comment (bkcsoft) [inline models/org_team.go]: Should probably have it's own check? Since some usernames are valid as team-names

Comment (LefsFlarey) [inline models/org_team.go]: Hrm...

**Comment by LefsFlarey** [comment]
Pushing a rebased version to cover up my embarrassing errors.

File: models/org_team.go [hunk 2]
```diff
@@ -194,9 +194,25 @@ func (t *Team) RemoveRepository(repoID int64) error {
 	return sess.Commit()
 }
 
+func IsUsableTeamName(name string) (err error) {
```
Comment (bkcsoft) [inline models/org_team.go]: no need to name `err`, just go for `error`

File: models/org_team.go [hunk 3]
```diff
@@ -194,9 +194,25 @@ func (t *Team) RemoveRepository(repoID int64) error {
...
+	var names = []string{"new"}
```
Comment (LefsFlarey) [inline models/org_team.go]: Would a `var reservedTeamNames` be better? In terms of capitalizations I mean.

Comment (bkcsoft) [inline models/org_team.go]: yeah, but outside the function, since it doesn't change during runtime :smile:

**Pull request closed without merging**